src/EditorFeatures/Test/Diagnostics/TestDiagnosticAnalyzerService.cs

```diff
@@ -40,8 +42,9 @@ internal TestDiagnosticAnalyzerService(ImmutableArray<AnalyzerReference> hostAna
             HostAnalyzerManager hostAnalyzerManager,
             AbstractHostDiagnosticUpdateSource hostDiagnosticUpdateSource,
             Action<Exception, DiagnosticAnalyzer, Diagnostic> onAnalyzerException,
-            IDiagnosticUpdateSourceRegistrationService registrationService = null)
-            : base(hostAnalyzerManager, hostDiagnosticUpdateSource, registrationService ?? new MockDiagnosticUpdateSourceRegistrationService())
+            IDiagnosticUpdateSourceRegistrationService registrationService = null,
+            IAsynchronousOperationListener listener = null)
```

Comment:
I just tried and there are just way more unit test that is using this that doesnt require listener. only reason this test required the listener was since it didn't mock IDiagnosticAnalyzerService and rather tried to use real one, due to that, it required all these that are usually only required for integration tests.